oap-server/server-library/library-client/src/main/java/org/apache/skywalking/oap/server/library/client/elasticsearch/ElasticSearchClient.java

```diff
@@ -291,6 +292,30 @@ public SearchResponse search(String indexName, SearchSourceBuilder searchSourceB
         return client.search(searchRequest);
     }
 
+    public SearchResponse search(String[] indexNames, SearchSourceBuilder searchSourceBuilder) throws IOException {
+        String[] fullIndexNames = formatIndexNames(indexNames);
+        SearchRequest searchRequest = new SearchRequest(fullIndexNames);
+        searchRequest.types(TYPE);
+        searchRequest.source(searchSourceBuilder);
+        return client.search(searchRequest);
+    }
+
+    public String[] filterNotExistIndex(String[] fullIndexNames, String indName) throws IOException {
+        // if no wrap, it is impossible to remove elements
+        List<String> indexNameList = new ArrayList<>(Arrays.asList(fullIndexNames));
+        if (fullIndexNames.length > 0) {
+            List<String> existIndex = retrievalIndexByAliases(indName);
+            indexNameList.removeIf(indexName -> {
+                //only filter index name with xxxx-xxxx
+                if (indexName.contains("-")) {
```

Comment:
Which one doesn't contain `-`?

Reply:
for example `segment alarm_record`

Reply:
Use `Const#LINE`

Reply:
ACK

Reply:
server-core depends on server-client, so `Const#LINE` can not used in server-client module.

Reply:
Then you could provide a filter func as the parameter to do the filter works. This indicates this logic doesn't belong to this class.